src/keymaster-api.js

```diff
@@ -40,27 +40,33 @@ v1router.get('/wallet', async (req, res) => {
         const wallet = keymaster.loadWallet();
         res.json(wallet);
     } catch (error) {
-        res.status(500).send(error.toString());
+        res.status(500).send({ error: error.toString() });
     }
 });
 
 v1router.put('/wallet', async (req, res) => {
     try {
         const { wallet } = req.body;
+        if (!wallet) {
+            return res.status(400).send({ error: 'Wallet data is required' });
+        }
         const response = keymaster.saveWallet(wallet);
         res.json(response);
     } catch (error) {
-        res.status(500).send(error.toString());
+        res.status(500).send({ error: error.toString() });
     }
 });
 
 v1router.post('/wallet/new', async (req, res) => {
     try {
         const { mnemonic, overwrite } = req.body;
+        if (!mnemonic) {
```

Comment:
API should not check parameters. A null mnemonic is valid here (keymaster will generate a new random mnemonic)